ACPs/176-dynamic-evm-gas-limit-and-price-discovery-updates/README.md

```diff
@@ -0,0 +1,157 @@
+
+1. The windower mechanism used to determine the base fee of blocks can lead to outsized spikes in the gas price when there is a large block. This is because after a large block that uses all of its gas limit, blocks that follow in the same window continue to result in increased gas prices even if they are relatively small blocks that are under the target gas consumption.
+2. The static gas target necessitates a required network upgrade in order to modify. This is cumbersome and makes it difficult for the network to adjust its capacity in response to performance optimization or hardware requirement increases.
+
+To better position Avalanche EVM chains, including the C-Chain, to be able to handle future increases in load, we propose replacing the above mechanism with one that better handles blocks that consume a large amount of gas, and that allows for validators to dynamically adjust the target rate of consumption.
+
+## Specification
+
+### Gas Price Determination
+
+The mechanism to determine the base fee of a block is the same as the one used in ACP-103 to determine the gas price of a block on the P-Chain. This mechanism calculates the gas price for a given block $b$ based on the following parameters:
+
+<div align="center">
+
+| | |
+|---|---|
+| $T$ | the target gas consumed per second |
+| $M$ | minimum gas price |
+| $K$ | gas price update constant |
+| $C$ | maximum gas capacity |
+| $R$ | gas capacity added per second |
+
+</div>
+
+The full specification of this mechanism can be found [here](../103-dynamic-fees/README.md).
+
+### Making $T$ Dynamic
+
+As noted above, the gas price determination mechanism relies on a target gas consumption per second, $T$, in order to calculate the gas price for a given block. $T$ will be adjusted dynamically according to the following specification.
+
+Let $q$ be a non-negative integer that is initialized to 0 upon activation of this mechanism. Let the target gas consumption per second be expressed as:
+
+$$T = P \cdot e^{q/D}$$
+
+where $P$ is the global minimum allowed target gas consumption rate for the network, and $D$ is a constant that helps control the rate of change of the target gas consumption.
+
+After the execution of transactions in block $b$, the values $q$ can be increased or decreased up to $Q$. It must be the case that $q_{b-1} - Q \leq q_{b} \leq q_{b-1} + Q$, or block $b$ is considered invalid. The amount by which $q$ changes after executing block $b$ is specified by the block builder.
```

Comment:
```suggestion
After the execution of transactions in block $b-1$, the values $q$ can be increased or decreased up to $Q$, or block $b$ will be considered invalid. The amount by which $q$ changes after executing block $b$ is specified by the block builder.
```
Both sentences are saying the same thing, and I think the first sentence is the most clear.

Reply:
I don't think these are the same actually. The builder of block $b$ gets to specify the change in $q$ that takes effect after the transactions in the block are executed, meaning it effects the gas price of transactions starting in block $b+1$. I added the extra specification just to make the block validation rule crystal clear.